src/fundus/parser/data.py

```diff
@@ -85,7 +88,7 @@ def get_value_by_key_path(self, key_path: List[str], default: Any = None) -> Opt
             tmp = nxt
         return tmp
 
-    def bf_search(self, key: str, depth: Optional[int] = None, default: Any = None) -> Optional[Any]:
+    def bf_search(self, key: str, depth: Optional[int] = None, default: Optional[_T] = None) -> Union[Any, _T]:
```

Comment:
Why does it make sense to update the return value to `Union[Any, _T]`?

Reply:
So mypy does know that if a default is given, at least that type is returned.

Reply:
Ah ok. I didn't know mypy was able to make that connection. Thanks